### ISSUE TITLE
can't read out Mekanism inventories with the processor

### ISSUE THREAD

**Issue by p4di** [McJty/RFToolsControl#45]
trying to read out and inventory from Mekanism with "eval: count items external" makes me get an internal error on that opcode. Reading out RF from a Mekanism Machine with the Processor works.

Since the RFTools possibilities like the "inventory module" for the screens or the "inventory checker" work fine with Mekanism inventories, I assume, that you maybe missed porting that function?

**Comment by McJty** [comment]
Not really. This looks more like a bug on Mek's side. Are you using a internal 'side' to access the mek machines?

**Comment by p4di** [comment]
I defined the opcode to read the inventory of the block on the west side, which is where I placed down the Mek machine. Defining a inventory slot to read out makes no difference in  the error.

Using the program with a chest instead works fine, reading out RF from the Mek machine with a similar program works fine as well.

As I said, using the inventory checker or a screen module from RFTools instead, is working.

Could it be, that it matters which side of the mek machine faces the west side of the processor? I should probably also check, whether nodes could do the job, I only tried it with the processor.

**Comment by p4di** [comment]
So I tried it again, this time telling the program to access the machine from the front and at least I'm not getting an error anymore, but reading out the inventory correctly still doesn't work, it always tells me there are 0 items in the inventory or slot.

Maybe there's still something wrong with my program, I've added some pictures.

![program](https://cloud.githubusercontent.com/assets/24624556/21431798/1187bc4a-c869-11e6-92d5-2b25720c9a8f.png)
![console](https://cloud.githubusercontent.com/assets/24624556/21431805/174be868-c869-11e6-9b6b-f45c83562185.png)
![other](https://cloud.githubusercontent.com/assets/24624556/21431810/1b976bc2-c869-11e6-93f4-908ff50b0e0e.png)

**Comment by McJty** [comment]
You are using W/S to access the items. Can you just try W/* instead?

**Comment by p4di** [comment]
W/* gives me the "internal Error" for 1/1

**Comment by McJty** [comment]
Well that *is* a bug in Mekanism as it seems not to accept 'null' for side for the item handler and this is required. You can report that to mekanism.

As to why it isn't working otherwise I also suspect Mekanism as rftools control just uses the basic API from forge/MC for accessing items

**Comment by p4di** [comment]
alright, I'll report it there, thanks for your help and this great mod!

**Comment by TheBeesElise** [comment]
I've been playing around with Mekanism's I/O mechanics. Mek machines only report the inventory slots associated with the side configuration you've assigned. If you have 'n' output slots in a machine, you can only read output slots slot from sides assigned to output, and it will return as slot 0 through 'n-1'. Same for input, energy, etc. You can't read from an unassigned side and you can only read the assigned slots.

**Comment by p4di** [comment]
that's an interesting discovery, I'll check out whether that works, thank you!

**Comment by p4di** [comment]
@DungeonMaestro thank you, I just tried and it works just fine, when you consider the side configuration!
